ThrottleInputApp 2/src/app/services/quix.service.ts

```diff
@@ -0,0 +1,259 @@
+  public workspaceId: string = 'demo-simulinkcombustionengine-frontend'; // Look in the URL for the Quix Portal your workspace ID is after 'workspace='
+  public topicName: string = 'throttle-input'; // get topic name from the Topics page in the Quix portal
+  /* optional */
+  /*~-~-~-~-~-~-~-~-~-~-~-~-~-~-~-*/
+
+  private subdomain = 'platform'; // leave as 'platform'
+  readonly server = ''; // leave blank
+
+  private readerReconnectAttempts: number = 0;
+  private writerReconnectAttempts: number = 0;
+  private reconnectInterval: number = 5000;
+  private hasReaderHubListeners: boolean = false;
+
+  public readerHubConnection: HubConnection;
+  public writerHubConnection: HubConnection;
+
+  private readerConnStatusChanged = new Subject<ConnectionStatus>();
+  readerConnStatusChanged$ = this.readerConnStatusChanged.asObservable();
+  private writerConnStatusChanged = new Subject<ConnectionStatus>();
+  writerConnStatusChanged$ = this.writerConnStatusChanged.asObservable();
+
+  paramDataReceived = new Subject<ParameterData>();
+  paramDataReceived$ = this.paramDataReceived.asObservable();
+
+  eventDataReceived = new Subject<EventData>();
+  eventDataReceived$ = this.eventDataReceived.asObservable();
+
+  private domainRegex = new RegExp(
+    "^https:\\/\\/portal-api\\.([a-zA-Z]+)\\.quix\\.ai"
+  );
+
+  constructor(private httpClient: HttpClient) {
+
+    if (this.workingLocally || location.hostname === "localhost" || location.hostname === "127.0.0.1") {
+      this.setUpHubConnections(this.workspaceId);
+    }
+    else {
+      const headers = new HttpHeaders().set('Content-Type', 'text/plain; charset=utf-8');
+      let bearerToken$ = this.httpClient.get(this.server + 'bearer_token', { headers, responseType: 'text' });
+      let workspaceId$ = this.httpClient.get(this.server + 'workspace_id', { headers, responseType: 'text' });
+      let portalApi$ = this.httpClient.get(this.server + 'portal_api', { headers, responseType: 'text' })
+      let topic$ = this.httpClient.get(this.server + 'throttle', { headers, responseType: 'text' });
```

Comment:
the name of the environment variable is a bit cryptic. Let's use `output_topic`.